### PR TITLE
OpcodeDispatcher: Implement support for the various prefetch instructions

### PR DESCRIPTION
**Pull request by Sonicadvance1**
x86 has a few prefetch instructions.
- prefetch - One of two classic 3DNow! instructions
   - Prefetch in to L1 data cache
- prefetchw - One of two classic 3DNow! instructions
   - Implies prefetch in to L1 data cache
   - Prefetch cacheline with intent to write and exclusive ownership

- prefetchnta
   - Prefetch non-temporal data in respect to /all/ cache levels
   - Assumes inclusive caches?
- prefetch{t0,t1,t2}
   - Prefetch data with respect to each cache level
   - T0 = L1 and higher
   - T1 = L2 and higher
   - T2 = L3 and higher

**Some silly duplicates**
- prefetchwt1
   - Duplicate of prefetchw but explicitly L1 data cache
- prefetch_exclusive
   - Duplicate of prefetch

God Of War 2018 uses prefetchw as a hint for exclusive ownership of the
cacheline in some very aggressive spin-loops. Let's implement the
operations to help it along.

### CODE REVIEW

**Comment by alyssarosenzweig** [comment]
> God Of War 2018 uses prefetchw as a hint for exclusive ownership of the cacheline in some very aggressive spin-loops.

Any perf numbers you can share? I'd generally expect real testing for an optimization that strictly adds instructions :stuck_out_tongue_winking_eye:

**Review by alyssarosenzweig — APPROVED** [review]
Code looks good :+1:

**Comment by Sonicadvance1** [comment]
> > God Of War 2018 uses prefetchw as a hint for exclusive ownership of the cacheline in some very aggressive spin-loops.
> 
> Any perf numbers you can share? I'd generally expect real testing for an optimization that strictly adds instructions 😜

Perf didn't seem to move in God Of War, but its worker thread CPU utilization was up ~10%. Can't tell if this was because it was doing more work, or if the work wasn't visible because the game was bottlenecked from the dxvk-cs thread.

**Comment by alyssarosenzweig** [comment]
> > > God Of War 2018 uses prefetchw as a hint for exclusive ownership of the cacheline in some very aggressive spin-loops.
> > 
> > 
> > Any perf numbers you can share? I'd generally expect real testing for an optimization that strictly adds instructions 😜
> 
> Perf didn't seem to move in God Of War, but its worker thread CPU utilization was up ~10%. Can't tell if this was because it was doing more work, or if the work wasn't visible because the game was bottlenecked from the dxvk-cs thread.

That's... uninspiring :|

File: unittests/InstructionCountCI/FlagM/SecondaryGroup.json [hunk 1]
```diff
@@ -1639,36 +1639,44 @@
       ]
     },
     "prefetchnta [rax]": {
-      "ExpectedInstructionCount": 0,
+      "ExpectedInstructionCount": 1,
       "Comment": [
         "GROUP16 0x0F 0x18 /0",
         "NOP implementation"
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply unittests/InstructionCountCI/FlagM/SecondaryGroup.json]: Oop yes, missed those comments. fixed.

**Review by alyssarosenzweig — APPROVED** [review]

**Pull request merged**